File: Data Structures & Algorithms/two-integer-sum/submission-0.go
func twoSum(nums []int, target int) []int {
    mp := make(map[int]int)

    for i, x := range nums {
        need := target - x

        if idx, ok := mp[need]; ok {
            return []int{idx, i}
        }

        mp[x] = i
    }

    return []int{-1, -1}
}

File: Data Structures & Algorithms/two-integer-sum-ii/submission-0.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& a, int target) {
        int n = a.size();

        int l = 0 , r = n-1;
        while(l<r){
            int sum = a[l] + a[r];
            if(sum == target) return {l+1,r+1};
            else if(target > sum) l++;
            else r--;
        }

        return {-1,-1};
    }
};
